test/cli/pubsub_test.go

```diff
@@ -0,0 +1,390 @@
+		nodes.ForEachPar(enablePubsub)
+		nodes = nodes.StartDaemons().Connect()
+
+		node1 := nodes[0]
+		node2 := nodes[1]
+		node2PeerID := node2.PeerID().String()
+
+		const topic = "seqno-test"
+
+		// Start subscriber on node1
+		go func() {
+			node1.RunIPFS("pubsub", "sub", topic)
+		}()
+		time.Sleep(500 * time.Millisecond)
+
+		// Publish multiple messages from node2 to trigger seqno validation
+		for i := 0; i < 3; i++ {
+			node2.PipeStrToIPFS("msg", "pubsub", "pub", topic)
+			time.Sleep(100 * time.Millisecond)
+		}
+
+		// Give time for messages to propagate and seqno to be stored
+		time.Sleep(1 * time.Second)
+
+		// Stop daemons to check datastore (diag datastore requires daemon to be stopped)
+		nodes.StopDaemons()
+
+		// Check that seqno state exists using diag datastore count
+		res := node1.IPFS("diag", "datastore", "count", "/pubsub/seqno/")
+		count := strings.TrimSpace(res.Stdout.String())
+		t.Logf("seqno entries count: %s", count)
+
+		// There should be at least one seqno entry (from node2)
+		assert.NotEqual(t, "0", count, "expected seqno state to be persisted")
+
+		// Verify the specific peer's key exists
+		key := "/pubsub/seqno/" + node2PeerID
+		res = node1.RunIPFS("diag", "datastore", "get", "--hex", key)
+		if res.Err == nil {
+			t.Logf("seqno for peer %s:\n%s", node2PeerID, res.Stdout.String())
+			assert.Contains(t, res.Stdout.String(), "Hex Dump:")
+		} else {
+			// Key might not exist if messages didn't propagate - log but don't fail
+			t.Logf("seqno key not found for peer %s (messages may not have propagated)", node2PeerID)
+		}
+	})
+
+	t.Run("seqno updates when receiving multiple messages", func(t *testing.T) {
+		t.Parallel()
+		h := harness.NewT(t)
+
+		// Create two connected nodes with pubsub
+		nodes := h.NewNodes(2).Init()
+		nodes.ForEachPar(enablePubsub)
+		nodes = nodes.StartDaemons().Connect()
+
+		node1 := nodes[0]
+		node2 := nodes[1]
+		node2PeerID := node2.PeerID().String()
+
+		const topic = "seqno-update-test"
+		seqnoKey := "/pubsub/seqno/" + node2PeerID
+
+		// Start subscriber on node1
+		go func() {
+			node1.RunIPFS("pubsub", "sub", topic)
+		}()
+		time.Sleep(500 * time.Millisecond)
+
+		// Send first message
+		node2.PipeStrToIPFS("msg1", "pubsub", "pub", topic)
+		time.Sleep(500 * time.Millisecond)
+
+		// Stop daemons to check seqno (diag datastore requires daemon to be stopped)
+		nodes.StopDaemons()
+
+		// Get seqno after first message
+		res1 := node1.RunIPFS("diag", "datastore", "get", seqnoKey)
+		var seqno1 []byte
+		if res1.Err == nil {
+			seqno1 = res1.Stdout.Bytes()
+			t.Logf("seqno after first message: %d bytes", len(seqno1))
+		} else {
+			t.Logf("seqno not found after first message (message may not have propagated)")
+		}
+
+		// Restart daemons for second message
+		nodes = nodes.StartDaemons().Connect()
+		time.Sleep(500 * time.Millisecond)
+
+		// Resubscribe
+		go func() {
+			node1.RunIPFS("pubsub", "sub", topic)
+		}()
+		time.Sleep(500 * time.Millisecond)
```

Comment:
Since there is an ability to wait for listeners (currently in separate PR), should that be used instead of using `time.Sleep`?